### ISSUE TITLE
win_reboot action logs connection tracebacks in some cases

### ISSUE THREAD

**Issue by nitzmahone** [ansible/ansible#17187]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

win_reboot action plugin
##### ANSIBLE VERSION

```
2.1 + 2.2 devel
```
##### SUMMARY

The win_reboot action internally makes direct connection/module runs that we expect to fail (we swallow the exceptions internally in these cases). However, in rare cases where the "canary" module exec fails, it appears that the module exec subsystem is directly logging/displaying the resultant traceback (even though it's being swallowed by the action). This can be alarming to users. We should either move the direct traceback logging out of the module subsystem entirely, or pass along an option to the module exec to suppress it.
##### STEPS TO REPRODUCE

```
Upgrade powershell on a slowish Windows box (AWS/Azure?) and reboot, for example.
```

##### EXPECTED RESULTS

Silent reboot and task success
##### ACTUAL RESULTS

Task success, but intermediate tracebacks are logged/displayed.

**Comment by ansibot** [comment]
@nitzmahone: Thanks for taking the time to open this issue, but it was filed under the wrong github repository. Please create a new issue at https://github.com/ansible/ansible-modules-core/issues
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jctanner** [comment]
sorry @nitzmahone, still ironing out the ansibot wrinkles.

**Comment by mattclay** [comment]
The [direct traceback writes](https://github.com/ansible/ansible/blob/4ed88512e45112f9670560ac3f01707a40a7f5c5/lib/ansible/plugins/connection/winrm.py#L276) are a result of the calls to `traceback.print_exc()` in the `winrm` connection plugin. This same pattern is used in most of the other connection plugins as well.

It would be a simple matter to remove these calls, but then the information would not be available in other situations when it is needed for troubleshooting. To preserve the original exception information we could:
- Use [exception chaining](http://python-future.org/compatible_idioms.html#raising-exceptions), which requires the `future` python module, or
- Pass the original traceback to the new exception we raise, which would require updating our existing exception classes.

Unfortunately, both of those changes are too invasive for a 2.1 or 2.2 release. Since this is only a cosmetic issue, I've removed the 2.2.0 milestone and we can revisit this issue later, hopefully for an eventual 2.3 release. Whatever the solution, it most likely should be applied to the other connection plugins as well.

**Comment by ansibot** [comment]
cc @dagwieers @jborean93 @jhawkesworth
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by nitzmahone** [comment]
This should be fixed in 2.5 once connection-oriented win_reboot rebuild is merged.

**Comment by jborean93** [comment]
win_reboot change was merged with https://github.com/ansible/ansible/pull/31767
